sdks/java/core/src/test/java/org/apache/beam/sdk/util/FileIOChannelFactoryTest.java

```diff
@@ -194,19 +207,23 @@ public void testMatchWithDirectoryFiltersOutDirectory() throws Exception {
 
   @Test
   public void testResolve() throws Exception {
-    String expected = temporaryFolder.getRoot().toPath().resolve("aa").toString();
-    assertEquals(expected, factory.resolve(temporaryFolder.getRoot().toString(), "aa"));
+    Path rootPath = temporaryFolder.getRoot().toPath();
+    String rootString = rootPath.toString();
+
+    String expected = rootPath.resolve("aa").toString();
+    assertEquals(expected, factory.resolve(rootString, "aa"));
+    assertEquals(expected, factory.resolve("file://" + rootString, "aa"));
```

Comment:
The former is a relative URL just as you think. The latter is invalid. If the authority is omitted the path portion is expressly forbidden from starting with `//`. That said, there are windows services that expect it...

Definitely should test them.